File: Java/Language/InterfaceTest2/Program2.java
import taxation.*;

class Program {
	
	//nested (static) member class can only reference static members
	//of its outer class
	static class Auditor implements AutoCloseable {

		public Auditor() {
			System.out.println("Acquiring resources");
		}

		public void audit(String id, TaxPayer target) {
			if(id.length() < 4)
				throw new IllegalArgumentException("Invalid ID");
			double payment = target.incomeTax() + 500;
			System.out.printf("Total Payment: %.2f%n", payment);
		}

		public void close() {
			System.out.println("Releasing resources");
		}

	}

	private static void process(String name, int count) {
		TaxPayer t = name.equals("jack") ? new Engineer(count) : new Programmer(count);
		try(Auditor a = new Auditor()){
			a.audit(name, t);
		}//compiler will attach finally block with a.close()
	}

	public static void main(String[] args) {
		try{
			String m = args[0].toLowerCase();
			int n = Integer.parseInt(args[1]);
			process(m, n);
		}catch(Exception e){
			System.out.printf("Error: %s%n", e);
		}
	}
}

File: dotnet/Language/Inheritance/InterfaceTest2/DemoApp/Program2.cs
using Taxation;

class Program
{
    //nested class - can only refer to static members of outer class
    class Auditor : IDisposable
    {
        internal Auditor()
        {
            Console.WriteLine("Acquiring resources.");
        }

        public void Audit(string id, ITaxPayer person)
        {
            if(id.Length < 4)
                throw new ArgumentException("Invalid ID");
            double payment = person.IncomeTax() + 500;
            Console.WriteLine("Total Payment: {0:0.00}", payment);
        }

        public void Dispose()
        {
            Console.WriteLine("Releasing resources.");
        }
    }

    private static void Process(string name, int count)
    {
        ITaxPayer t = name == "jack" ? new Engineer(count) : new Programmer(count);
        using(Auditor a = new Auditor())
        {
            a.Audit(name, t);
        }
    }

    public static void Main(string[] args)
    {
        try
        {
            string m = args[0].ToLower();
            int n = int.Parse(args[1]);
            Process(m, n);
        }
        catch(Exception ex)
        {
            Console.WriteLine($"Error: {ex.Message}");
        }
    }
}